sleep_constant_time.go

```diff
@@ -0,0 +1,52 @@
+package discordgo
+
+import (
+	"fmt"
+	"time"
+)
+
+type sleepCT struct {
+	d     time.Duration // desired duration between targets
+	t     time.Time     // last time target
+	wake  time.Time     // last wake time
+	drift int64         // last wake drift microseconds
+}
+
+func NewSleepCT(d time.Duration) sleepCT {
+
+	s := sleepCT{}
+
+	s.d = d
+	s.t = time.Now()
+
+	return s
+}
+
+func (s *sleepCT) SleepNext() int64 {
+
+	now := time.Now()
+
+	// if target is zero safety net
+	if s.t.IsZero() {
+		fmt.Println("TickerCT reset")
```

Comment:
We try to avoid logging outside of `Session`, so I think would be better to remove this message.

Reply:
of course thanks for your diligence. If you would like anything else changed let me know.